usmqe/usmscheduler.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf8 -*-
+"""
+Pytest plugin for scheduling tasks on nodes.
+.. moduleauthor:: Filip Balak <fbalak@redhat.com>
+"""
+import usmqe.usmssh as usmssh
+import datetime
+import random
+import string
+
+
+class Scheduler(object):
+    """
+    Wrapper around `usmssh` module that handles scheduling tasks on multiple
+    nodes simultaneously.
+    """
+
+    def __init__(self, nodes):
+        """
+        Args:
+            nodes (list): list of nodes used for command scheduling
+        """
+        self.nodes = nodes
+        self.ssh = usmssh.get_ssh()
+
+    def run_at(self, command, time=None):
+        """
+        Schedule a job on all machines with `at` command.
+
+        Args:
+            command (str): Command to be executed.
+            time (time): Time to execute. Time has to be in `at` compatible
+                format. If None is provided then all commands are executed
+                in next minute.
+        """
+        if not time:
+            time = datetime.datetime.now() + datetime.timedelta(minutes=1)
+            time = time.strftime("%H:%M")
+        files = self.create_job_file(command)
+        for node in self.nodes:
+            time_command = "at {0} -f {1} ".format(time, files[node])
+            self.ssh[node].run(time_command)
+
+    def create_job_file(self, command):
+        """
+        Create task file in /tmp directory which can be used by at command.
+
+        Args:
+            command (str): Command to be executed.
+
+        Returns:
+            dict: Keys are node hostnames and values are file paths.
+        """
+        files = {}
+        for node in self.nodes:
+            char_set = string.ascii_lowercase + string.digits
+            file_name = "/tmp/schedulertask_{0}".format(
+                ''.join(random.sample(char_set*6, 6)))
+            create_file_cmd = "echo '#!/bin/sh\n{0}' > {1}".format(
+                command.replace("'", "\\'"),
```

Comment:
I'm not sure if this is good enough ... why don't you need to escape just this one case?

Reply:
I dropped this char escaping. Python seems to handle this correctly and it seems that using both single and double quotes is working.

Reply:
ok